docs improvement (#8545)
crate/crate: blackbox/docs/general/builtins/table-functions.rst

Table Functions
===============

> Table functions are functions that produce a set of rows which are made up of

Reviewer: ```suggestion
Table functions are functions that produce a set of rows. They can either be used in place of a relation in the FROM clause, or within the select list of a query.

If used within the select list, the table functions will be evaluated per row ...
```

> either base data types (scalar types) or composite data types (table rows).

Reviewer: I'd revert this change. I don't understand this "either base data types (scalar types) or composite data types addition. I think this confuses more than it clarifies.

    around is not allowed, unless sub queries are utilized.
    (SELECT aggregate_func(col) FROM (SELECT table_func(...) as col) ...)

.. rubric:: Table of Contents

.. contents::
   :local:

.. _table-functions-scalar:

Scalar functions
================

> :ref:`scalar`, when used in the ``FROM`` clause of a query as a table or subquery,

Reviewer: What are you referring to with `as a table or subquery` ? Not sure if this makes more sense to list all types of possible relations and imply that the scalar can be used as such. I think the point is that the scalar can be used in the FROM and that it will act like any other relation would.

Author: > What are you referring to with as a table or subquery?
The proper term would be to use the `relation` abstraction. And, I agree that enumerating all different options is not nice, however, it would be inconsistent with other places I saw. I'm happy to change it there as well though and keep the relation abstraction.
 
> I think the point is that the scalar can be used in the FROM and that it will act like any other relation would.

The key point (and what I wanted to stress here) is the "like any other relation would").
I think it's an important point as a function can be used in the `From` clause as an expression and not as a relation eg in 
`from substr(initcap('hello world'),1, 5)` 
`substr` scalar will be treated as a relation while `initcap` wouldn't despite the fact that both are used in the query `From` clause.

My point was to highlight this difference.

> :ref:`scalar`, when used in the ``FROM`` clause of a query as a table or subquery,

Reviewer: ```suggestion
:ref:`scalar`, when used in the ``FROM`` clause in place of a relation, 
```
